File: 0056-merge-intervals/0056-merge-intervals.py
class Solution(object):
    def merge(self, intervals):
        intervals.sort(key=lambda x:x[0])
        arr=[]
        for i in intervals:
            if not arr or arr[-1][1]<i[0]:
                arr.append(i)
            else:
                arr[-1][1]=max(arr[-1][1],i[1])
        return arr

File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& a) {
        sort(a.begin(), a.end());
        vector<vector<int>> r;
        for (auto& x : a) {
            if (r.empty() || r.back()[1] < x[0])
                r.push_back(x);
            else
                r.back()[1] = max(r.back()[1], x[1]);
        }
        return r;
    }
};
